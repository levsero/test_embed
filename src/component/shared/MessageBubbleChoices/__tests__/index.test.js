import { render } from 'react-testing-library'
import React from 'react'

import { MessageBubbleChoices } from '../index'

test('renders the expected classes', () => {
  const { container } = render(
    <MessageBubbleChoices
      leadingMessage="this is the leading message"
      leadingMessageStyle="gangnamStyle"
      containerStyle="containerStyle"
    >
      <div className="firstChild" />
      <div className="secondChild" />
      <div className="thirdChild" />
    </MessageBubbleChoices>
  )

  expect(container).toMatchSnapshot()
})

test('renders the expected classes when there is only one child', () => {
  const { container } = render(
    <MessageBubbleChoices
      leadingMessage="this is the leading message"
      leadingMessageStyle="gangnamStyle"
      containerStyle="containerStyle"
    >
      <div className="onlyChild" />
    </MessageBubbleChoices>
  )

  expect(container).toMatchSnapshot()
})

test('renders the expected classes when there is no leading message', () => {
  const { container } = render(
    <MessageBubbleChoices>
      <div className="firstChild" />
      <div className="secondChild" />
    </MessageBubbleChoices>
  )

  expect(container).toMatchSnapshot()
})
