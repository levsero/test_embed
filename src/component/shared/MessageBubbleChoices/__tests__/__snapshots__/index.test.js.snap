// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected classes 1`] = `
<div>
  <div
    class="messageBubble
        containerStyle"
    data-testid="answer-bot-selection-message"
  >
    <div
      class="header firstItem gangnamStyle"
    >
      this is the leading message
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="firstChild"
      />
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="secondChild"
      />
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="thirdChild"
      />
    </div>
  </div>
</div>
`;

exports[`renders the expected classes when there is no leading message 1`] = `
<div>
  <div
    class="messageBubble
        "
    data-testid="answer-bot-selection-message"
  >
    <div
      class="firstItem optionItem"
    >
      <div
        class="firstChild"
      />
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="secondChild"
      />
    </div>
  </div>
</div>
`;

exports[`renders the expected classes when there is only one child 1`] = `
<div>
  <div
    class="messageBubble
        containerStyle"
    data-testid="answer-bot-selection-message"
  >
    <div
      class="header firstItem gangnamStyle"
    >
      this is the leading message
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="onlyChild"
      />
    </div>
  </div>
</div>
`;
